src/checkers/inference/model/AlwaysTrueConstraint.java

```diff
@@ -0,0 +1,26 @@
+package checkers.inference.model;
+
+import java.util.Collections;
+
+/**
+ * This "constraint" is the result of normalizing another constraint, where that constraint is
+ * always true (evaluates to tautology). If a constraint is normalized to this constraint, then it
+ * is not added to the set of constraints by the {@link ConstraintManager}.
+ * 
+ * @see {@link ConstraintManager}
+ */
+public class AlwaysTrueConstraint extends Constraint {
+
+    private AlwaysTrueConstraint() {
+        super(Collections.emptyList());
+    }
+
+    public static Constraint create() {
```

Comment:
For some Constraints its `create` method is `protected`, e.g. `CombineConstraint`, and for some others are `public`, like this one.

What is the motivation of having different visibilities of `create` methods in different Constraints?

Reply:
Good catch. All of them are supposed to be protected. Will fix.